diff --git a/node_modules/@strapi/admin/admin/src/pages/HomePage/index.js b/node_modules/@strapi/admin/admin/src/pages/HomePage/index.js
index f8b8d85..247e66f 100644
--- a/node_modules/@strapi/admin/admin/src/pages/HomePage/index.js
+++ b/node_modules/@strapi/admin/admin/src/pages/HomePage/index.js
@@ -77,10 +77,10 @@ const HomePage = () => {
           </Grid>
           <Grid gap={6}>
             <GridItem col={8} s={12}>
-              {showGuidedTour ? <GuidedTourHomepage /> : <ContentBlocks />}
+              {/* {showGuidedTour ? <GuidedTourHomepage /> : <ContentBlocks />} */}
             </GridItem>
             <GridItem col={4} s={12}>
-              <SocialLinks />
+              {/* <SocialLinks /> */}
             </GridItem>
           </Grid>
         </Box>
